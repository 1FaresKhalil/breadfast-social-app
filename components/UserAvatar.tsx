import React from 'react';
import { View, Image } from 'react-native';
import { styled } from 'nativewind';

interface UserAvatarProps {
  userId: number;
}

const StyledView = styled(View);
const StyledImage = styled(Image);

const UserAvatar: React.FC<UserAvatarProps> = ({ userId }) => {
  return (
    <StyledView className="w-10 h-10 bg-gray-300 rounded-full overflow-hidden">
      <StyledImage
        source={{ uri: `https://robohash.org/${userId}.png` }}
        className="w-full h-full"
      />
    </StyledView>
  );
};

export default UserAvatar;
