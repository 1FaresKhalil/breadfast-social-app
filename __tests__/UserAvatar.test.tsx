import React from 'react';
import renderer from 'react-test-renderer';
import UserAvatar from '../components/UserAvatar';

describe('<UserAvatar />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<UserAvatar userId={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
