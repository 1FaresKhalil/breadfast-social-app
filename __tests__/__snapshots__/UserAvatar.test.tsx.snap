// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserAvatar /> renders correctly 1`] = `
<View
  style={
    [
      [
        {
          "width": 40,
        },
        {
          "height": 40,
        },
        {
          "backgroundColor": "#d1d5db",
        },
        {
          "borderBottomLeftRadius": 9999,
          "borderBottomRightRadius": 9999,
          "borderTopLeftRadius": 9999,
          "borderTopRightRadius": 9999,
        },
        {
          "overflow": "hidden",
        },
      ],
    ]
  }
>
  <Image
    source={
      {
        "uri": "https://robohash.org/1.png",
      }
    }
    style={
      [
        [
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ],
      ]
    }
  />
</View>
`;
